src/components/operate.js

```diff
@@ -0,0 +1,22 @@
+function readEqual(state) {
+  if (state.firstNumber === ''
+    || state.secondNumber === ''
+    || state.operation === '') {
+    return 0;
+  }
+
+  if (state.operation === '+') {
+    return Number(state.firstNumber) + Number(state.secondNumber);
+  } if (state.operation === '-') {
+    return Number(state.firstNumber) - Number(state.secondNumber);
+  } if (state.operation === '×') {
+    return Number(state.firstNumber) * Number(state.secondNumber);
+  } if (state.operation === '÷') {
+    return Number(state.firstNumber) / Number(state.secondNumber);
+  } if (state.operation === '%') {
+    return Number(state.firstNumber) % Number(state.secondNumber);
+  }
+  return 0;
+}
+
+export default { readEqual };
```

Comment:
- [ ] Kindly note that it's really required to use the operation logic provided by the requirement. see [here](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/code-samples/operate.js)